pubtools/pulplib/_impl/page.py

```diff
@@ -87,12 +92,13 @@ def do_cancel(*_):
             self.__dict__["_cancel_ref"] = weakref.ref(self, do_cancel)
 
     def as_iter(self):
-        """Returns an iterator which individually yields each object in this
-        page, and all subsequent pages.
```

Comment:
Can we also add here some TODO, fox FIX or whatever keyword that could indicate that this should be removed at version x.y.z ?

Reply:
+1
Also, if planning to remove as_iter in later version, I would suggest to get that done sooner as this library has a relatively smaller usage across projects as of now. This will lead to minimal changes with backward incompatibility once as_iter is removed.

Reply:
I've added a TODO comment now.  I think removal of deprecated features should be time-based and not version-based (we discussed it on mailing list a while back, thread "Default deprecation period?"), so I record the approximate date at which it was deprecated rather than try to set now at which version it will be removed.